holoviews/util.py

```diff
@@ -127,3 +127,47 @@ def _make_dynamic(self, hmap, dynamic_fn, streams):
         kdims = [d(values=list(set(values))) for d, values in
                  zip(hmap.kdims, dim_values)]
         return DynamicMap(dynamic_fn, streams=streams, **dict(params, kdims=kdims))
+
+
+class ParamDimList(param.ParameterizedFunction):
+    """
+    Simple utility that accepts a parameterized object and returns a
+    corresponding list of dimension objects.
+    """
+
+    exclude = param.List(default=['name'], doc="""
+        Parameters to exclude by name from the dimension list.""")
+
+    label = param.Boolean(default=True, doc="""
+        Whether to use the parameter documentation as the dimension label""")
+
+    precedence_threshold = param.Number(default=0, doc="""
+        The threshold below which parameters are ignored.""")
+
+    def __call__(self, parameterized, **kwargs):
+
+        types = {param.String:str, param.Integer:int}
+        p = param.ParamOverrides(self, kwargs)
+        dim_list = []
+        params = parameterized.params()
+        for name, param_obj in params.items():
+            if name in p.exclude:  continue
+            elif param_obj.precedence:
+                if param_obj.precedence < p.precedence_threshold:
+                    continue
+            dim_kws = {}
+            if param_obj.doc and p.label:
+                dim_kws['label'] = param_obj.doc
+            if isinstance(param_obj, param.Number):
+                if param_obj.bounds != None:
+                    dim_kws['range'] = param_obj.bounds
+                if param_obj.softbounds != None:
+                    dim_kws['soft_range'] = param_obj.softbounds
+            elif isinstance(param_obj, param.List):
+                dim_kws['values'] = param_obj.default
```

Comment:
This doesn't make sense to me. A ``param.List`` expects the whole list as input but this treats it basically as an ObjectSelector.